File: models/Producto.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Producto = sequelize.define('producto', {
  id_producto: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true,
    field: 'id_producto'
  },
  nombre: {
    type: DataTypes.STRING,
    allowNull: false,
    unique: true
  },
  descripcion: {
    type: DataTypes.STRING,
    allowNull: false
  },
  precio_compra: {
    type: DataTypes.FLOAT,
    allowNull: false,
    field: 'precio_compra'
  },
  precio_venta: {
    type: DataTypes.FLOAT,
    allowNull: false,
    field: 'precio_venta'
  },
  stock: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  active: {
    type: DataTypes.BOOLEAN,
    defaultValue: true
  }
}, {
  tableName: 'productos',
  timestamps: true,
  underscored: true
});

module.exports = Producto;

File: lib/models/producto.dart
class Producto {
  final int idProducto;
  final String nombre;
  final String descripcion;
  final double precioCompra;
  final double precioVenta;
  final int stock;
  final bool active;

  Producto({
    required this.idProducto,
    required this.nombre,
    required this.descripcion,
    required this.precioCompra,
    required this.precioVenta,
    required this.stock,
    required this.active,
  });

  factory Producto.fromJson(Map<String, dynamic> json) {
    return Producto(
      idProducto: json['id_producto'] ?? json['idProducto'],
      nombre: json['nombre'],
      descripcion: json['descripcion'],
      precioCompra: (json['precio_compra'] ?? json['precioCompra']).toDouble(),
      precioVenta: (json['precio_venta'] ?? json['precioVenta']).toDouble(),
      stock: json['stock'],
      active: json['active'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_producto': idProducto,
      'nombre': nombre,
      'descripcion': descripcion,
      'precio_compra': precioCompra,
      'precio_venta': precioVenta,
      'stock': stock,
      'active': active,
    };
  }
}